src/js/game/LevelMVC/LevelModel.js

```diff
@@ -781,11 +781,19 @@ module.exports = class LevelModel {
         break;
     }
 
-    if (shouldPlace === true) {
-      var block = new LevelBlock(blockType);
+    if (entity === this.agent) {
+      let ground = this.groundPlane.getBlockAt(position);
+      if (ground.blockType === "water" || ground.blockType === "lava") {
+        this.groundPlane._data[this.groundPlane.coordinatesToIndex(position)] = new LevelBlock(blockType);
```

Comment:
use `setBlockAt` rather than `_data`

Reply:
I wanted to say there was some reason for this... Either that it was done that way before (in which case copy-paste and you're totally right) or that for some reason setBlockAt() caused a problem... I wanted to say setBlockAt() was being weird recently with something, but this might not be the PR. Will replace this logic and if anything funky happens I'll ping you.

Reply:
Oh, it's because setBlockAt() does a number of things that presume it's the action plane. That's probably an erroneous presumption, so I'll look into refactoring that first 👍